File: src/collectors/sources/SedailySourceAdapter.ts
import { BaseRSSAdapter } from "./BaseRSSAdapter.js";

/**
 * 서울경제 RSS 수집 어댑터.
 * 경제 전문 언론사로, 산업·금융·부동산 뉴스를 폭넓게 다룹니다.
 */
export class SedailySourceAdapter extends BaseRSSAdapter {
  constructor() {
    super({
      sourceName: "서울경제",
      sourceType: "news_media",
      feedUrls: [
        "https://www.sedaily.com/RSS/Economy",
      ],
    });
  }
}

File: src/main/java/com/economicbriefing/collector/source/SedailySourceAdapter.java
package com.economicbriefing.collector.source;

import java.util.List;

import com.economicbriefing.collector.filter.CategoryClassifier;
import com.economicbriefing.collector.parser.ArticleNormalizer;
import com.economicbriefing.collector.parser.RssParser;
import com.economicbriefing.domain.article.ArticleSourceType;
import com.economicbriefing.domain.article.NewsCategory;
import org.springframework.stereotype.Component;

@Component
public class SedailySourceAdapter extends AbstractRssSourceAdapter {

    public SedailySourceAdapter(RssParser rssParser, ArticleNormalizer normalizer, CategoryClassifier categoryClassifier) {
        super(rssParser, normalizer, categoryClassifier);
    }

    @Override
    public String getSourceName() { return "서울경제"; }

    @Override
    public String getFeedUrl() { return "https://www.sedaily.com/RSS/Economy"; }

    @Override
    public ArticleSourceType getSourceType() { return ArticleSourceType.NEWS_MEDIA; }

    @Override
    public List<NewsCategory> getDefaultCategories() { return List.of(); }
}
